modules/v2/variables.tf

```diff
@@ -0,0 +1,313 @@
+  default     = {}
+}
+
+variable "client" {
+  type = object({
+    name    = optional(string, null)
+    version = optional(string, null)
+  })
+  description = "Arbitrary identifier for the API client and version identifier"
+  default     = {}
+}
+
+variable "ingress" {
+  type        = string
+  description = "Provides the ingress settings for this Service. On output, returns the currently observed ingress settings, or INGRESS_TRAFFIC_UNSPECIFIED if no revision is active."
+  default     = "INGRESS_TRAFFIC_ALL"
+
+  validation {
+    condition     = contains(["INGRESS_TRAFFIC_ALL", "INGRESS_TRAFFIC_INTERNAL_ONLY", "INGRESS_TRAFFIC_INTERNAL_LOAD_BALANCER"], var.ingress)
+    error_message = "Allowed values for ingress are \"INGRESS_TRAFFIC_ALL\", \"INGRESS_TRAFFIC_INTERNAL_ONLY\", or \"INGRESS_TRAFFIC_INTERNAL_LOAD_BALANCER\"."
+  }
+}
+
+variable "launch_stage" {
+  type        = string
+  description = "The launch stage as defined by Google Cloud Platform Launch Stages. Cloud Run supports ALPHA, BETA, and GA. If no value is specified, GA is assumed."
+  default     = "GA"
+
+  validation {
+    condition     = contains(["UNIMPLEMENTED", "PRELAUNCH", "EARLY_ACCESS", "ALPHA", "BETA", "GA", "DEPRECATED"], var.launch_stage)
+    error_message = "Allowed values for launch_stage are \"UNIMPLEMENTED\", \"PRELAUNCH\", or \"EARLY_ACCESS\", or \"DEPRECATED\", or \"ALPHA\", or \"BETA\", or \"GA\"."
+  }
+}
+
+variable "custom_audiences" {
+  type        = list(string)
+  description = "One or more custom audiences that you want this service to support. Specify each custom audience as the full URL in a string. Refer https://cloud.google.com/run/docs/configuring/custom-audiences"
+  default     = null
+}
+
+variable "binary_authorization" {
+  type = object({
+    breakglass_justification = optional(bool) # If present, indicates to use Breakglass using this justification. If useDefault is False, then it must be empty. For more information on breakglass, see https://cloud.google.com/binary-authorization/docs/using-breakglass
+    use_default              = optional(bool) #If True, indicates to use the default project's binary authorization policy. If False, binary authorization will be disabled.
+  })
+  description = "Settings for the Binary Authorization feature."
+  default     = null
+}
+
+// Template
+variable "revision" {
+  description = "The unique name for the revision. If this field is omitted, it will be automatically generated based on the Service name"
+  type        = string
+  default     = null
+}
+
+variable "template_scaling" {
+  type = object({
+    min_instance_count = optional(number)
+    max_instance_count = optional(number)
+  })
+  description = "Scaling settings for this Revision."
+  default     = {}
+}
+
+variable "vpc_access" {
+  type = object({
```

Comment:
Should this be list(object())? I can see this is being used in a dynamic block with intention of creating multi `vpc_access`.

Reply:
No. This is a single block.